### PR TITLE
Moved validator code into main

### PR DESCRIPTION
**Pull request by Anwesh1**
Moved the validator code into `main.py` and made the required changes. Now we need to implement a way in which it only calls `normalize_documents()` if the id provided isn't present in the `indexed_ids`.

https://github.com/PathwayCommons/semantic-search/blob/1d4bbc50e6b090deddddd58ccdef392b7b00e5d3/semantic_search/common/util.py#L122-L126

However, this causes an issue because FAISS doesn't seem to have the ability to retrieve vectors if their id is provided, which is what we need to do.

### CODE REVIEW

**Comment by JohnGiorgi** [comment]
> However, this causes an issue because FAISS doesn't seem to have the ability to retrieve vectors if their id is provided, which is what we need to do.

I am confused, why do we need to retrieve vectors by their ids?

**Comment by Anwesh1** [comment]
`indexed_ids` is a set that holds all the `ids` in the index. All of these ids have a corresponding vector that has been encoded from the id's text. It is stored in the index as a vector so the text is essentially gone after it has been encoded, I believe. If we can still retrieve the text then that would be better because I was thinking of structuring it like this:
```python
if int(query.query.uid) not in indexed_ids:
        if query.query.text is None: 
            query.query.text = normalize_documents([query.query.uid])
else:

```

**Comment by JohnGiorgi** [comment]
I believe there is a much simpler solution. Can you look at the changes I pushed? I tested that if I make the same request twice, with PMIDs only, the second request is many times faster (suggesting that we are not fetching the text because it is already indexed).

File: semantic_search/main.py [hunk 1]
```diff
@@ -99,24 +99,17 @@ async def query(query: Query):
     # Only add items to the index if they do not already exist.
     # See: https://github.com/facebookresearch/faiss/issues/859
     # To do this, we first determine which of the incoming ids do not exist in the index
+    indexed_ids = set(faiss.vector_to_array(model.index.id_map).tolist())
 
-    if query.query.text is None:
+    if query.query.text is None and query.query.id not in indexed_ids:
```
Comment (Anwesh1) [inline semantic_search/main.py:102]: I think it should be `query.query.uid ` just misspelled I think.

File: semantic_search/main.py [hunk 2]
```diff
@@ -97,8 +98,16 @@ async def query(query: Query):
     # See: https://github.com/facebookresearch/faiss/issues/859
     # To do this, we first determine which of the incoming ids do not exist in the index
     indexed_ids = set(faiss.vector_to_array(model.index.id_map).tolist())
-    to_embed = [(id_, text) for id_, text in zip(ids, texts) if id_ not in indexed_ids]
+
+    if query.query.text is None and query.query.id not in indexed_ids:
```
Comment (JohnGiorgi) [inline semantic_search/main.py:102]:
```suggestion
    if query.query.text is None and query.query.uid not in indexed_ids:
```

**Comment by JohnGiorgi** [comment]
@Anwesh1 Cool! There is a test breaking but it is solved in #68 so ill fix it there.

You don't have to apologize! Everything I mentioned is supposed to be taken as advice/tips/guidance, not scolding!

**Pull request merged**